src/redis-cli.c

```diff
@@ -1591,19 +1491,19 @@ static int parseOptions(int argc, char **argv) {
             double seconds = atof(argv[++i]);
             config.interval = seconds*1000000;
         } else if (!strcmp(argv[i],"-n") && !lastarg) {
-            config.input_dbnum = atoi(argv[++i]);
+            config.conn_info.input_dbnum = atoi(argv[++i]);
         } else if (!strcmp(argv[i], "--no-auth-warning")) {
             config.no_auth_warning = 1;
         } else if (!strcmp(argv[i], "--askpass")) {
             config.askpass = 1;
         } else if ((!strcmp(argv[i],"-a") || !strcmp(argv[i],"--pass"))
                    && !lastarg)
         {
-            config.auth = argv[++i];
+            config.conn_info.auth = argv[++i];
```

Comment:
God knows if we need to free the existing `auth` and whether it's a `char*` and `sds` or a pointer to the environment returned by `getenv` which we can't free... I guess this was always here, so you decide if you want to handle this...

Reply:
it's ugly, but i'm not sure it's a real issue.
i.e. the memory leak is not a severe one (one that keeps accumulating), so i don't mind living with it.
although, you are right, that it's probably nicer to always copy the data to an sds, so that we always need to release it in the same way.